opennms-services/src/main/java/org/opennms/netmgt/poller/pollables/LatencyStoringServiceMonitorAdaptor.java

```diff
@@ -110,88 +111,83 @@ private void storeResponseTime(MonitoredService svc, Map<String, Number> entries
+            applyThresholds(collectionSet, svc, dsName, repository);
+        } else {
+            LOG.debug("storeResponseTime: Thresholds processing is not enabled. Check thresholding-enabled parameter on service definition");
+        }
+
+        LOG.debug("storeResponseTime: Persisting latency data for {}", svc);
+        persistCollectionSet(collectionSet, repository);
     }
 
-    private void persistLatencySamples(MonitoredService service, Map<String, Number> entries, File rrdRepositoryRoot, String rrdBaseName) {
+    private RrdRepository getRrdRepository(String rrdPath) {
+        File rrdRepositoryRoot = new File(rrdPath);
         RrdRepository repository = new RrdRepository();
         repository.setStep(m_pollerConfig.getStep(m_pkg));
         repository.setRraList(m_pollerConfig.getRRAList(m_pkg));
         repository.setHeartBeat(repository.getStep() * HEARTBEAT_STEP_MULTIPLIER);
         repository.setRrdBaseDir(rrdRepositoryRoot);
+        return repository;
+    }
+
+    private void persistCollectionSet(CollectionSet collectionSet, RrdRepository repository) {
+        ServiceParameters params = new ServiceParameters(Collections.emptyMap());
+        CollectionSetVisitor persister = m_persisterFactory.createPersister(params, repository, false, true, true);
+        collectionSet.visit(persister);
+    }
+
+    private void applyThresholds(CollectionSet collectionSet, MonitoredService service, String dsName, RrdRepository repository) {
+        try {
+            if (m_thresholdingSession == null) {
+                m_thresholdingSession = m_thresholdingService.createSession(service.getNodeId(), 
+                                                                            service.getIpAddr(),
+                                                                            service.getSvcName(),
+                                                                            repository,
+                                                                            EMPTY_SERVICE_PARAMS);
+            }
+            m_thresholdingSession.accept(collectionSet);
+        } catch (Throwable e) {
+            LOG.error("Failed to threshold on {} for {} because of an exception", service, dsName, e);
+        }
+    }
 
+    private CollectionSet getCollectionSet(MonitoredService service, Map<String, Number> entries, String rrdBaseName) {
         // When making calls directly to RrdUtils#createRrd() and RrdUtils#updateRrd(),
         // the behavior was as follows:
         // 1) All samples get written to response/${ipAddr}/${rrdBaseName}.rrd
         //     This happens whether or not storeByGroup is enabled.
         // 2) If multiple entries are present, the DSs are created in the same order that they
         //    appear in the map
 
-        LatencyCollectionResource latencyResource = new LatencyCollectionResource(service.getSvcName(), service.getIpAddr(), service.getNodeLocation());
+        String ifLabel = "";
```

Comment:
This doesn't seem like the right place for this. Why are we collecting interface info for a latency collection set here?

Reply:
please see threshd/CollectionResourceWrapper.java in the diff
In the early refactoring, I wanted to pull it up from the CollectionResourceWrapper creation, so Thresholding would have to know less about the callers. However there is still a check on ResourceType so I can easily put it back.